architecture/replication-layer.md

```diff
@@ -0,0 +1,77 @@
+---
+title: Replication Layer
+summary: 
+toc: false
+---
+
+<div id="toc"></div>
+
+## Overview
+
+To ensure survivability, CockroachDB replicates all ranges at least 3 times. This allows for one range to become inaccessible without compromising the consistency of the system.
+
+Up to F failures can be tolerated, where the total number of replicas N = 2F + 1 (e.g. with 3x replication, one failure can be tolerated; with 5x replication, two failures, and so on).
+
+However, the number of replicas is ultimately controlled with table-level granularity in your cluster's zone config file.
+
+CockroachDB ensures consistency on these replicas––despite being distributed––by using the Raft consensus algorithm.
+
+## Components
+
+### Raft
+
+Raft works by organizing all nodes that contain a replica of a range into a group––unsurprisingly called a Raft Group.
+
+Raft elects a relatively long-lived leader which must be involved to propose commands. It heartbeats followers periodically and keeps their logs replicated. In the absence of heartbeats, followers become candidates after randomized election timeouts and proceed to hold new leader elections. Cockroach weights random timeouts such that the replicas with shorter round trip times to peers are more likely to hold elections first (not implemented yet). Only the Raft leader may propose commands; followers will simply relay commands to the last known leader.
+
+Once a node receives `BatchRequests` for a range it contains, it converts those KV operations into Raft commands. Those commands are proposed to the Raft group leader––which is what makes it ideal for the Lease Holder and the Raft leader to be one in the same; it reduces network roundtrips if the Lease Holder receiving the requests can simply propose the Raft commands to itself, rather than communicating them to another node.
```

Comment:
From here and below I don't understand exactly what the lease holder does. Does the lease holder ALWAYS receive the first request and then, assuming it is a write, forward it to the raft leader? cc @bdarnell